src/software/thunderscope/robot_diagnostics/drive_and_dribbler_widget.py

```diff
@@ -16,32 +24,61 @@ def __init__(self, proto_unix_io: ProtoUnixIO) -> None:
         self.input_a = time.time()
         self.constants = tbots_cpp.create2021RobotConstants()
         QWidget.__init__(self)
+
         layout = QVBoxLayout()
+        self.drive_widget = QStackedWidget()
 
         self.proto_unix_io = proto_unix_io
 
-        # Add widgets to layout
-        layout.addWidget(self.setup_direct_velocity("Drive"))
-        layout.addWidget(self.setup_dribbler("Dribbler"))
+        # create swappable widget system using stacked widgets
+        self.direct_velocity_widget = self.setup_direct_velocity(
+            "Drive - Direct Velocity"
+        )
+        self.per_motor_widget = self.setup_per_motor("Drive - Per Motor")
+        self.drive_widget.addWidget(self.direct_velocity_widget)
+        self.drive_widget.addWidget(self.per_motor_widget)
 
-        self.enabled = True
+        layout.addWidget(self.setup_drive_switch_radio("Drive Mode Switch"))
+        layout.addWidget(self.drive_widget)
+        layout.addWidget(self.setup_dribbler("Dribbler"))
 
+        # default to direct velocity
+        self.drive_mode = DriveMode.VELOCITY
+        self.use_direct_velocity.setChecked(True)
+        self.toggle_drive_mode(DriveMode.VELOCITY)
```

Comment:
I believe you could replace this with `self.use_direct_velocity.click()` since it is attached with the QtSignal to `toggle_drive_mode`